test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -292,6 +294,74 @@ public void StringSegment_EqualsString_Invalid()
             Assert.False(result);
         }
 
+        public static TheoryData<StringSegment, StringComparison> DefaultStringSegmentEqualsStringSegmentData
+        {
+            get
+            {
+                // candidate / comparer / expected result
+                return new TheoryData<StringSegment, StringComparison>()
+                {
+                    { default(StringSegment), StringComparison.Ordinal},
+                    { new StringSegment(), StringComparison.Ordinal},
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(DefaultStringSegmentEqualsStringSegmentData))]       
+        public void DefaultStringSegment_EqualsStringSegment(StringSegment candidate, StringComparison comparison)
+        {
+            // Arrange
+            var segment = default(StringSegment);
+           
+            // Act
+            var result = segment.Equals(candidate, StringComparison.Ordinal);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Fact]
+        public void DefaultStringSegment_DoesNotEqualStringSegment()
```

Comment:
Could you convert this into a theory and add a test data with `new StringSegment(string.Empty)` and `new StringSegment("Hello", 1, 0)`?